functions-and-scope.js

```diff
@@ -0,0 +1,207 @@
+// * Hoe kan ik iedere waarde van de array checken op deze conditie?
+// * Hoe zorg ik ervoor dat dit ook werkt wanneer de array 100 entries bevat?
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan bijhouden?
+// Log het antwoord in de terminal.
+
+// Stappenplan:
+// 1. Variabele maken voor aantal cumlaude studenten
+// 2. For loop maken voor het doorlopen van de array van cijfers, check hierbij de lengte van de array als referentie voor de duur van de loop
+// 3. Checken of cijfer hoger of gelijk is dan 8
+// 4. Wanneer hoger of gelijk dan 8 +1 op cumlaude studenten variabele
+// 5. Log resultaat van cumlaude studenten variabele naar console
+
+let cumLaudeGraduates = 0
+
+for(let i = 0; i != grades.length; i++)
+{
+    if (grades[i] >= 8 )
+    {
+        cumLaudeGraduates++
+    }
+}
+
+console.log(cumLaudeGraduates)
+
+
+// ---- Verwachte uitkomst: 6
+
+
+/*  1b: Omschrijven tot een herbruikbare functie   */
+// Schrijf een functie genaamd cumLaude, die een array van cijfers verwacht (zoals grades) en het aantal Cum laude studenten teruggeeft. Gebruik hiervoor jouw antwoord van 1a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array met eindcijfers willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Log het antwoord in de terminal.
+
+function cumLaude(grades){
+    let cumLaudeGraduates = 0
+
+    for(let i = 0; i != grades.length; i++)
+    {
+        if (grades[i] >= 8 )
+        {
+            cumLaudeGraduates++
+        }
+    }
+
+    return cumLaudeGraduates
+}
+console.log(cumLaude(grades))
+console.log(cumLaude([6,4,5]))
+console.log(cumLaude([8, 9, 4, 6, 10]))
+
+// ---- Verwachte uitkomsten:
+// cumLaude(grades) geeft 6
+// cumLaude([6, 4, 5]) geeft 0
+// cumLaude([8, 9, 4, 6, 10]) geeft 3
+
+
+
+
+/* Opdracht  2: Gemiddeld cijfer */
+
+/* 2a: Script schrijven  */
+// De studenten-administratie moet ieder blok opnieuw berekenen wat het gemiddelde eindcijfer is, maar we beginnen met de grades array van hierboven.
+// Schrijf de stapjes om dit te kunnen berekenen eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe wordt een gemiddelde berekend?
+// * Wat moet ik verzamelen uit de array van cijfers om uiteindelijk een gemiddelde te kunnen berekenen?
+// * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten?
+// Log het antwoord in de terminal.
+
+// ---- Verwachte uitkomst: 6.642857142857143
+
+// Stappenplan
+// 1. Maak een variabele voor totaal van alle cijfers
+// 2. Tel alle waardes in de array bij elkaar op en stop dit in de eerder aangemaakte variabele
+// 3. Deel de uitkomst van de variabele door de lengte van de array en sla het resultaat op in een variabele
+// 4. Output de uitkomst via een console.log
+
+let totalGrades = 0
+
+for(let i = 0; i != grades.length; i++)
+{ 
+    totalGrades = totalGrades + grades[i]
```

Comment:
Dit kun je versimpelen:
```suggestion
    totalGrades += grades[i];
```

Reply:
Ah ja, thanks